### ISSUE TITLE
Roslyn build restrictions

### ISSUE THREAD

**Issue by jaredpar** [dotnet/roslyn#20929]
There are a number of requirements Roslyn has on the path on which it is enlisted:

- Cannot contain spaces
- Cannot contain SEMICOLON (U+003B) character
- Cannot contain `%` character

These limitations are unnecessary and in theory could be supported. However it would require **both**

- Changes to our build scripts and external tooling to support the characters
- Changing to our test infrastructure to verify we don't regress the behavior

The latter in particular is the most important. There have been a number of attempts to remove our limitation on no spaces. These have never been accompanied with the corresponding infrastructure changes and hence the fixes last a week at most before we break them again.

**Comment by VladimirReshetnikov** [comment]
Also:
* Path cannot contain `$(foo)`
* Path cannot contain `@(foo)`

**Comment by jaredpar** [comment]
Added those.

**Comment by jasonmalinowski** [comment]
@jaredpar Given the number of bugs that have came up over time, and the number of pull requests we've gotten from customers who are actually eager to fix these issues, any reason we shouldn't just fix these and update CI to put a space in the path? This seems particularly unfriendly to open source contributors who are either eager to make their first contributions, or those who might be on corporate owned machines who might have no choice.

**Comment by jaredpar** [comment]
> any reason we shouldn't just fix these and update CI to put a space in the path? 

I'm 100% okay with doing that. I'm against taking any fixes to spaces, or other restrictions, though until we do this. Otherwise it puts us in a mixed state and causes friction.

**Comment by sharwell** [comment]
Can we make Build.cmd fail early with a reasonable message if the path doesn't meet requirements?

**Comment by jaredpar** [comment]
Yes that's pretty easy to do. I'm actually following up ATM with AzDev to see if there is a way to control the path name in CI. This was doable in Jenkins not sure how in AzDev.

**Comment by jasonmalinowski** [comment]
@jaredpar What was the expected ETA on that?
